backend/src/app.ts

```diff
@@ -8,18 +8,58 @@ const app: Application = express()
 interface Tournament {
     id: string;
     name: string;
-    'joining-code': number;
+    player_code: string;
+    admin_code: string;
 }
 
-app.set('port', (process.env.PORT || 5000));
+
+app.use(express.urlencoded({ extended: false }));
+
+app.set('port', (process.env.PORT || 8000));
 
 app.get('/tournaments', async(req: Request, res: Response) => {
-    const query = pg.select("id", "name").from<Tournament>("tournament");
+    const query = pg.select("id", "name", "player_code", "admin_code").from<Tournament>("tournament");
     const tournaments = await query;
     
     res.json( { tournaments });
 });
 
+app.post('/tournaments', async(req: Request, res: Response) => {
+    const body = req.body;
+
+    for (let requiredParameter of ['name', 'player_code', 'admin_code']) {
+        if (!body[requiredParameter]) {
+          return res
+            .status(422)
+            .send({ error: `Expected format: { name: <String>, player_code: <String>, admin_code: <String> }. You're missing a "${requiredParameter}" property.` });
+        }
+    }
+    
+    try {
+    await pg("tournament").insert(body);
+    res.send("Created!")
```

Comment:
You should use `res.json` like in the GET endpoint and return the data of the created tournament. It's especially important that you include the generated ID in the response so that the client JavaScript code knows where to navigate to.

Reply:
Done